src/core/bsequence.ml

```diff
@@ -0,0 +1,49 @@
+
+type +'a node =
+    | Nil
+    | Cons of 'a * 'a t
+and 'a t = unit -> 'a node
+
+let empty () = Nil
+
+let return x () = Cons(x, empty)
```

Comment:
ah, I think we might be agreed then. I don't really care about the terminology, but I think the interface is very valuable, both the beginners and more experienced devs alike.

As long as there is a standard to adhere to that is consistent inside this repo, I can mentally replace bind and return with make and flatMap or andThen or whatever the best name is decided to be.